subgraphs/aave-forks/src/mapping.ts

```diff
@@ -372,707 +511,590 @@ export function _handleReserveDataUpdated(
     return;
   }
 
-  const totalBorrowBalance = sBorrowBalance
-    .plus(vBorrowBalance)
-    .toBigDecimal()
-    .div(exponentToBigDecimal(inputToken.decimals));
-  market.totalBorrowBalanceUSD = totalBorrowBalance.times(assetPriceUSD);
-
   // update total supply balance
-  const aTokenContract = AToken.bind(Address.fromString(market.outputToken!));
+  const aTokenContract = AToken.bind(Address.fromBytes(market.outputToken!));
   const tryTotalSupply = aTokenContract.try_totalSupply();
   if (tryTotalSupply.reverted) {
     log.warning(
       "[ReserveDataUpdated] Error getting total supply on market: {}",
-      [marketId.toHexString()]
+      [market.id.toHexString()]
     );
     return;
   }
+
+  if (assetPriceUSD.equals(BIGDECIMAL_ZERO)) {
+    assetPriceUSD = market.inputTokenPriceUSD;
+  }
+  manager.updateMarketAndProtocolData(
+    assetPriceUSD,
+    tryTotalSupply.value,
+    vBorrowBalance,
+    sBorrowBalance
+  );
+
   const tryScaledSupply = aTokenContract.try_scaledTotalSupply();
   if (tryScaledSupply.reverted) {
     log.warning(
       "[ReserveDataUpdated] Error getting scaled total supply on market: {}",
-      [marketId.toHexString()]
+      [asset.toHexString()]
     );
     return;
   }
 
-  market.inputTokenBalance = tryTotalSupply.value;
-  market.outputTokenSupply = tryTotalSupply.value;
-  market.totalDepositBalanceUSD = market.inputTokenBalance
-    .toBigDecimal()
-    .div(exponentToBigDecimal(inputToken.decimals))
-    .times(assetPriceUSD);
-  market.totalValueLockedUSD = market.totalDepositBalanceUSD;
-
   // calculate new revenue
   // New Interest = totalScaledSupply * (difference in liquidity index)
+  if (!market._liquidityIndex) {
+    market._liquidityIndex = BIGINT_ONE;
```

Comment:
Should really be set to RAY: https://github.com/aave/protocol-v2/blob/master/contracts/protocol/libraries/logic/ReserveLogic.sol#L173

Reply:
good catch! fixed